### PR TITLE
migrate to webpack 4 and mini-css-extract-plugin

### PR DESCRIPTION
**Pull request by JounQin**

### CODE REVIEW

**Comment by arthurmuchir** [comment]
It looks MiniCssExtractPlugin does not currently supports SSR (problem with `document`) :
https://github.com/webpack-contrib/mini-css-extract-plugin/issues/48

**Comment by JounQin** [comment]
Async style injection is handled by `vue-style-loader`.

**Comment by 132yse** [comment]
emmm，Not MiniCssExtractPlugin does not supports SSR，but Vue-loader does not support MiniCssExtractPlugin that it only supports ExtractTextPlugin，How to do should I？？？

**Comment by JounQin** [comment]
@132yse just use vue-loader@next (15.0+) instead which supports MiniCssExtractPlugin

**Comment by 132yse** [comment]
(⊙o⊙)…我升级了vue-loader，然而……仿佛是不能支持stylus-loader的::>_<::
看上去还是比较ok的
谢啦
（不好意思……英语不符合我性格(⊙o⊙)…）

File: build/webpack.base.config.js [hunk 1]
```diff
@@ -47,38 +50,42 @@ module.exports = {
...
-                  options: { minimize: true }
-                },
-                'stylus-loader'
-              ],
-              fallback: 'vue-style-loader'
-            })
-          : ['vue-style-loader', 'css-loader', 'stylus-loader']
+        oneOf: [
+          {
+            test: /App/,
```
Comment (FranckFreiburger) [inline build/webpack.base.config.js]: please can you give some explanations about this test ? thanks

**Pull request closed without merging**